app/src/main/java/com/alphawallet/app/entity/tokens/ERC721Ticket.java

```diff
@@ -61,12 +63,7 @@ public String getFullBalance() {
 
     @Override
     public Map<BigInteger, NFTAsset> getTokenAssets() {
-        Map<BigInteger, NFTAsset> assets = new HashMap<>();
-        for (BigInteger tokenId : balanceArray)
-        {
-            assets.put(tokenId, new NFTAsset(tokenId));
-        }
-        return assets;
+        return balanceArray.stream().collect(Collectors.toMap(tokenId -> tokenId, NFTAsset::new, (a, b) -> b));
```

Comment:
Please revert to 'for' loop. This is a bit impenetrable what's going on in here, would never be sure exactly what this does without expanding the code back out.